### PR TITLE
Added single node topological traversal and quickcheck tests.

### LINKED ISSUE #63: Topological iterator for subgraph.
It would be nice if there was way to traverse directed graph topologically starting from certain node.

### PR DESCRIPTION
**Pull request by WaDelma**
Resolves #63

### CODE REVIEW

**Comment by bluss** [comment]
This algorithm should work well. (I've never read about it, I wonder if it's the usual way to do it?)

I'm reluctant to add these extra datastructures to Topo. It's better as a separate walker. You can name it SubgraphTopo if you want. Or SubTopo or so?

**Comment by WaDelma** [comment]
Yeah I was also iffy about adding more stuff to `Topo`, but splitting is also not ideal as well as it creates code dublication.    
I tried to search if there was some other way to do it, but didn't find anything.

File: tests/quickcheck.rs [hunk 1]
```diff
@@ -453,3 +457,194 @@ fn graph_sccs() {
     }
     quickcheck::quickcheck(prop as fn(_) -> bool);
 }
+
+#[derive(Debug, Clone)]
+struct DAG<N: Default + Clone + Send + 'static>(Graph<N, ()>);
+
+impl<N: Default + Clone + Send + 'static> quickcheck::Arbitrary for DAG<N> {
```
Comment (bluss) [inline tests/quickcheck.rs]: We don't need to fix it now, but N should probably just be Arbitrary and generated by qc.

**Comment by bluss** [comment]
Quickcheck too, that's nice!

Can we cut down the number of constructors for SubTopo?

File: tests/quickcheck.rs (hunk 1 above)
Comment (WaDelma) [inline tests/quickcheck.rs]: Well the nodes value is used by the quicktest for both generic `Topo` and `SubTop` currently.

**Comment by WaDelma** [comment]
Shoulde the constructor then be just named `new` and take the node as parameter?

**Comment by bluss** [comment]
from node is good, then we can expand later if it's needed.

**Comment by WaDelma** [comment]
Well I already renamed it :P Will I change it back?

**Comment by bluss** [comment]
please use from_node

**Comment by bluss** [comment]
Just for future reference, github does not notify for new commits, needs comments on the PR.

**Comment by bluss** [comment]
Oops, sorry, my merging of the other PR conflicted with this branch. Didn't think of it.

**Comment by bluss** [comment]
Hm. This algorithm really has some interesting behavior for cyclic graphs. Unlike Topo, it will traverse nodes that participate in a cycle, where the whole cycle is reachable from the start node. That's a bit problematic. Can we solve that with documentation alone?

**Comment by WaDelma** [comment]
Rebased.    
Yeah should add that to documentation.

**Comment by WaDelma** [comment]
Whoops

**Comment by WaDelma** [comment]
I added note that cycles are included in `SubTopo`.

**Comment by bluss** [comment]
Thank you! Let's gooo. I'll look at this locall and maybe edit something, then petgraph is ready for a new release.

**Pull request merged**